extension/js/background.js

```diff
@@ -35,18 +35,33 @@ var salr_client = false;
  * within the extension.
  *
  */
-var port = chrome.runtime.connect();
-
-port.onMessage.addListener(function(data) {
-    if (!salr_client) {
-        salr_client = new SALR(data, chrome.extension.getURL("images/"));
-    }
-});
+var port = createPort();
 
 // Request the settings from the extension
 port.postMessage({'message': 'GetPageSettings'});
 
+// Create a port to communicate through.
+function createPort() {
+    let p = chrome.runtime.connect();
+    p.onMessage.addListener(function(data) {
+        if (!salr_client) {
+            salr_client = new SALR(data, chrome.extension.getURL("images/"));
+        }
+    });
```

Comment:
Given the other review comment, I'd actually move this listener outside of this function altogether and only bind it to the initial value of `port` as it should live long enough to get the response back from SALR, and that would give slightly less overhead on any additional ports we create. On that vein, I'd actually maybe rework it to be:

```
var port = createPort();
p.onMessage.addListener(function init(data) {
  salr_client = new SALR(data, chrome.extension.getURL("images/"));
  p.onMessage.removeListener(init);
});
```
where once we initialize `salr_client` variable, we can just rid of the listener altogether on the original instance, and also get rid of the `if` statement that was inside of it.

Thoughts?